### DISCUSSION TITLE
Welcome to selling-partner-api-samples Discussions!

### DISCUSSION THREAD

**Discussion by shreeharsh-a** [amzn/selling-partner-api-samples#35] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to Hot Topics with our community. We hope that you:
  * Let us know what do you think about the Hot Topics in the comments.
  * Share what do you want to know more next.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

If you have a question that you would like to post, please feel free to add them to Issues.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by rodrifed** [comment]
Federico here! 👋🏻
I am a Solutions Architect at Amazon. I enjoy interacting with the developer community in GitHub and events.
Looking forward to meeting y'all!

**Comment by leonardonakagawa** [comment]
Hey all!

I'm Naka! A Solutions Architect at Amazon. Happy to engage with you all! 
Excited to exchange ideas =)

**Comment by johnkw** [comment]
Hi. At first glance it's unfortunate that `selling-partner-api-docs` open bugs/discussions were moved to `selling-partner-api-models` and not here. This seems like a more logical successor.

Perhaps they should be moved again?  I don't know what the purpose of the `selling-partner-api-models` repo is though.  (The actual github code in that repo seems to be pointless json duplication of the docs.)  But that's the repo where people are filing bug reports on the API itself and the API documentation now, so it's the place to go for status reports.

**Comment by chenlianga123** [comment]
**putItemListing API Call error:** 
     Based on the data from '[fulfillment_channel_availability#?.inventory#?.inventory_type]', the field '"quantity"' for the attribute 'fulfillment_availability' is not allowed. Expected at most '0' of field '"quantity"' for attribute 'fulfillment_availability'.

**Found after debugging**
Adding a single property requires passing in this field（fulfillment_availability.is_inventory_available）
the new increment property cannot be passed into this field（fulfillment_availability.is_inventory_available）
Otherwise it will fail.

I would like to know which fields need to be transmitted by different distribution channels in different scenarios

**looking forward to a reply**

**Reply by johnkw** [reply]
That deserves a stand-alone "Issues" report, not a comment in this repository discussion.  Also you should link to the documentation page you're referencing for any issue.

**Reply by rodrifed** [reply]
Hi @chenlianga123,

The `quantity` attribute is accepted under `fulfillment_availability` when **fulfillment_channel_code = DEFAULT** (i.e. MFN / Fulfilled by Merchant).
For FBA fulfillment channel (**fulfillment_channel_code = AMAZON_NA / _EU / _FE**), `quantity` can't be specified as this information is already known by Amazon, based on the units sent to Amazon warehouse.

I would suggest to look at the [Manage Advanced Multiple-Offer and Multiple-Fulfillment Use Cases guide](https://developer-docs.amazon.com/sp-api/docs/manage-advanced-multiple-offer-multiple-fulfillment-use-cases), which will help you understand the requirements for different fulfillment channels.

Thank you,
Federico
Selling Partner Developer Services

**Comment by lmarkenson** [comment]
Hello @rodrifed and @leonardonakagawa,

I have a question about using the JSON_LISTING_FEED to update quantity and price.

If I use the PATCH call with the REPLACE operation, can I use the generic product type 'PRODUCT' regardless of the current 'product type' value of the listing on Amazon?

Thank you, Leo.

**Comment by rodrifed** [comment]
Hi @lmarkenson,

That is correct.
Because pricing and inventory attributes are part of the generic product type 'PRODUCT' (you can confirm this by calling the [/getDefinitionsProductType](https://developer-docs.amazon.com/sp-api/docs/product-type-definitions-api-v2020-09-01-reference#getdefinitionsproducttype) operation with `productType = PRODUCT`), you can use it for this purpose while calling the PATCH opertion.

Best,
Federico
Selling Partner Developer Services

**Reply by lmarkenson** [reply]
Hello @rodrifed 

Thank you for your quick response.

I encountered an error when I attempted to submit a JSON_LISTING_FEED with operationType set to "PARTIAL_UPDATE" and productType set to "PRODUCT" for a field (e.g., item_name, cost_price) that exists in the "PRODUCT" productType schema.

Below is the error I encountered:

{
  "issues": [
    {
      "messageId": 1,
      "code": "4005037",
      "severity": "ERROR",
      "message": "The specified listing was not found and cannot be edited. Create the listing before attempting to edit it.",
      "attributeName": "product_type"
    }

When I submitted the same feed using the productType value from the active listing on Amazon, it worked successfully.

Could you confirm if the behavior I described above is expected? If so, are there any additional differences I should be aware of between a PATCH with REPLACE operation and a feed with operationType set to "PARTIAL_UPDATE"? 

Thank you very much for your assistance.

Best regards,
Leo

**Reply by weilinggu** [reply]
Hi @lmarkenson,

We have tried using PARTIAL_UPDATE with PRODUCT product type and it worked for us. However, what works for us might not be applicable to your specific use case.
We suggest you [open a support case](https://developer.amazonservices.com/support) following the [guidance](https://developer.amazonservices.com/support-guidelines) here to troubleshoot what exactly was the issue of your submission first. Make sure to provide your submission request and response in your support case to speed up the troubleshooting process.

Thanks,
Weiling
Selling Partner API Developer Services
